perfmetrics/scripts/hns_rename_folders_metrics/generate_folders_and_files.py

```diff
@@ -95,7 +95,85 @@ def list_directory(path) -> list:
     return contents_url
   except subprocess.CalledProcessError as e:
     logmessage(e.output.decode('utf-8'))
-    subprocess.call('bash', shell=True)
+
+
+def compare_folder_structure(folder, folder_url) -> bool:
+  """Checks if the number of files in config file under folder matches the
+  number of files in the GCS bucket.
+
+  Args:
+    folder: Json Object representing the folder.
+    folder_url: Corresponding folder url in the GCS bucket.
+
+  Returns:
+    true if the number of files in the folder in GCS bucket matches the num_files
+    parameter of the folder JSON object.
+    false otherwise
+  """
+  try:
+    files_in_folder = list_directory(folder_url)
+    if len(files_in_folder) != folder["num_files"]:
+      return False
+  except:
+    # If the list directory fails wth url did not match object, folder
+    # specified in config file does not exist in bucket.
+    return False
+
+  return True
+
+
+def check_if_dir_structure_exists(directory_structure) -> bool:
+  """Checks if the directory structure mentioned in the config file already
+  exists in the GCS bucket.
+
+  Args:
+    directory_structure: Json Object representing the directory structure.
+
+  Returns:
+    true if the existing structure in GCS bucket exactly matches with teh config
+     file
+    false otherwise
+  """
+  bucket_name = directory_structure["name"]
+  bucket_url = 'gs://{}'.format(bucket_name)
+
+  # Check for top level folders.
+  folders = list_directory(bucket_url)
+  nested_folder_count = "nested_folders" in directory_structure
```

Comment:
Can't this check also go in function?

Reply:
Logic is different in this case as we have to handle cases where config file might not have a nested folder section.